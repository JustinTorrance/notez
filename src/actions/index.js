export const createNote = (note) => ({
  type: 'CREATE_NOTE',
  note
});

export const getNotes = (notes) => ({
  type: 'GET_NOTES',
  notes
});

export const isLoading = (boolean) => ({
  type: 'IS_LOADING',
  isLoading: boolean
});

export const fetchErrored = (message) => ({
  type: 'FETCH_ERRORED',
  message
});

export const deleteNoteAction = (id) => ({
  type: 'DELETE_NOTE',
  id
});

export const toggleCompleted = (id) => ({
  type: 'TOGGLE_COMPLETED',
  id
})