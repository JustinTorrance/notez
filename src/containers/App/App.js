import React, { Component } from 'react';
import Input from '../Input/Input'

class App extends Component {
  constructor() {
    super()
    this.state = {
      inputs: [],
      title: '',
      body: []
    }
  }

createNewItem = (text) => {
  const newListItem = { id: Date.now(), text}
  const newInput =  <Input key={Date.now()} createNewItem={this.createNewItem} />
  const body = [...this.state.body, newListItem]
  const inputs = [...this.state.inputs, newInput]
  this.setState({ body, inputs })
}

handleTitleChange = (e) => {
  this.setState({ title: e.target.value })
}

  render() {
    const mappedState = this.state.inputs.map(form => {
      return form
    })
    return (
      <div className="App">
        <h1>Howdy!</h1>
        <input type="text"
          placeholder='title'
          value={this.state.title}
          onChange={this.handleTitleChange}
        />

        <Input createNewItem={this.createNewItem} />
        
        {mappedState}
      </div>
    );
  }
}

export default App;