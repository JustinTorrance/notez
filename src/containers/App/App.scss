body {
  background-color: #fafafa;
  font-family: Source Sans Pro, sans-serif;
  margin: 0;
  padding: 0;
}

header {
  background-color: black;
  color: white;
  display: flex;
  height: 5rem;
  text-align: center;
}

h1 {
  margin: auto auto;
  padding: 0;
  justify-content: center;
  align-content: center;
}