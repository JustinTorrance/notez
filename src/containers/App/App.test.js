import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps } from './App';
import { isLoading } from '../../actions';

describe('App', () => {
  let wrapper;
  let mockProps = {
    loading: false
  };

  beforeEach(() => {
    wrapper = shallow(<App loading={mockProps.loading} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should match the snapshot if loading is true', () => {
    const mockProps = {
      loading: true
    };
    expect(wrapper.debug()).toMatchSnapshot();
  });
  
  describe(mapStateToProps, () => {
    it('should return a boolean that indicates whether isLoad is true or false', () => {
      const mockState = {
        isLoading: false
      }
      const expected = {
        loading: false
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })
});
