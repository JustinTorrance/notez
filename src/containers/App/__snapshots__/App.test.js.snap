// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
"<ContextConsumer>
  [function bound indirectRenderWrappedComponent]
</ContextConsumer>"
`;

exports[`App should match the snapshot if loading is true 1`] = `
"<ContextConsumer>
  [function bound indirectRenderWrappedComponent]
</ContextConsumer>"
`;
